cdap-ui/app/cdap/components/DataPrep/Directives/Parse/Modals/SingleFieldModal.js

```diff
@@ -30,9 +30,28 @@ export default class SingleFieldModal extends Component {
     };
 
     this.onTextChange = this.onTextChange.bind(this);
+    this.onOptionalTextChange = this.onOptionalTextChange.bind(this);
     this.apply = this.apply.bind(this);
   }
 
+  componentDidMount() {
+    this.allMouseTrapInstances = [];
+    document
+      .querySelectorAll('.modal .modal-content .form-control')
+      .forEach(ele => this.allMouseTrapInstances.push(new MouseTrap(ele)));
+    this.allMouseTrapInstances
+      .forEach(mtInstance => mtInstance.bind('enter', this.apply));
+    MouseTrap.bind('enter', this.apply);
```

Comment:
ahh, mousetrap does not execute if focus is on form element. You need to add a class `mousetrap` to the form element.

ref: https://craig.is/killing/mice#api.bind.text-fields